pennylane/ops/op_math/condition.py

```diff
@@ -430,3 +446,128 @@ def wrapper(*args, **kwargs):
+    def _(*_, jaxpr_true, jaxpr_false, jaxpr_elifs):
+
+        # We check that the return values in each branch (true, and possibly false and elifs)
+        # have the same abstract values.
+        # The error messages are detailed to help debugging
+        def validate_abstract_values(
+            outvals: list, expected_outvals: list, branch_type: str, index: int = None
+        ) -> None:
+            """Ensure the collected abstract values match the expected ones."""
+
+            assert len(outvals) == len(expected_outvals), (
+                f"Mismatch in number of output variables in {branch_type} branch"
+                f"{'' if index is None else ' #' + str(index)}: "
+                f"{len(outvals)} vs {len(expected_outvals)}"
+            )
+            for i, (outval, expected_outval) in enumerate(zip(outvals, expected_outvals)):
+                assert outval == expected_outval, (
+                    f"Mismatch in output abstract values in {branch_type} branch"
+                    f"{'' if index is None else ' #' + str(index)} at position {i}: "
+                    f"{outval} vs {expected_outval}"
+                )
+
+        outvals_true = jaxpr_true.out_avals
+
+        if jaxpr_false is not None:
+            outvals_false = jaxpr_false.out_avals
+            validate_abstract_values(outvals_false, outvals_true, "false")
+
+        for idx, jaxpr_elif in enumerate(jaxpr_elifs):
+            outvals_elif = jaxpr_elif.out_avals
+            validate_abstract_values(outvals_elif, outvals_true, "elif", idx)
+
+        # We return the abstract values of the true branch since the abstract values
+        # of the false and elif branches (if they exist) should be the same
+        return outvals_true
+
+    return cond_prim
+
+
+def _capture_cond(condition, true_fn, false_fn=None, elifs=()) -> Callable:
+    """Capture compatible way to apply conditionals."""
+
+    import jax  # pylint: disable=import-outside-toplevel
+
+    cond_prim = _get_cond_qfunc_prim()
+
+    elifs = (elifs,) if len(elifs) > 0 and not isinstance(elifs[0], tuple) else elifs
+
+    @wraps(true_fn)
+    def new_wrapper(*args, **kwargs):
+
+        jaxpr_true = jax.make_jaxpr(functools.partial(true_fn, **kwargs))(*args)
+        jaxpr_false = (
+            (jax.make_jaxpr(functools.partial(false_fn, **kwargs))(*args) if false_fn else None)
+            if false_fn
+            else None
+        )
+
+        # We extract each condition (or predicate) from the elifs argument list
+        # since these are traced by JAX and are passed as positional arguments to the cond primitive
+        elifs_conditions = []
+        jaxpr_elifs = []
+
+        for pred, elif_fn in elifs:
+            elifs_conditions.append(pred)
+            jaxpr_elifs.append(jax.make_jaxpr(functools.partial(elif_fn, **kwargs))(*args))
+
+        elifs_conditions = (
+            jax.numpy.array(elifs_conditions) if elifs_conditions else jax.numpy.empty(0)
+        )

```

Comment:
Why is this needed, just for convenience?

Reply:
`elifs` conditions are traced (as the first argument of the `qml.cond` function). Therefore, we pass them as a positional argument the the primitive, but if we try to pass them as a `list` or a `tuple` the execution fails because these are not standard JAX types.
This is why we have to turn such conditions into a JAX array. Am I missing something?

Reply:
Ah got it. Well there are other ways to pass multiple positional variadic arguments, you just have to unpack them (`*elifs_conditions`) in the function call. The only downside is on the other end of the call (in the function definition), you would have to separate out the multiple variadic arguments from a single list of values again. You can see patterns like this in the Catalyst primitives:
```py
(condition, *elif_conditions_and_args, elif_jaxprs=()):
    elif_conditions = elif_conditions_and_args[:len(elif_jaxprs)]
    args = elif_conditions_and_args[len(elif_jaxprs):]
```

But the be honest I don't know what is better yet so I'm not requesting any changes :)